.github/workflows/ci-for-default-ns.yaml

```diff
@@ -75,15 +75,21 @@ jobs:
           # for testing start-debug and stop-debug deployment
           # mons
           kubectl rook_ceph debug start rook-ceph-mon-a
-          sleep 5
+          sleep 10
+          debug_mon=$(kubectl get pod -l app=rook-ceph-mon,app.kubernetes.io/instance=a -n rook-ceph -o jsonpath='{.items[*].metadata.name}')
```

Comment:
What's to stop this line from returning the old pod if it doesn't terminate immediately? Often, pods take a while to terminate. Instead, can we select just the debug pod? The name of the deployment is deterministic (`rook-ceph-mon-a-debug` IIRC) also, so we should be able to use that to our advantage.

Reply:
A reusable function for waiting for a deployment's pod to be running is probably a good idea. I see this overall structure 3 times in this PR.

I use this script for my development environment tooling.

`wait_for`:
```sh
#!/usr/bin/env bash
set -eEuo pipefail

REASON="$1"  # reason
TIMEOUT="$2"  # in seconds
CMD="$3"

BASH_CMD="${BASH_CMD:=bash}" # might be 'bash -x', for example

msg="Waiting ${TIMEOUT} seconds for ${REASON} ..."
echo "${msg} countdown: ${TIMEOUT}"
start_time=$SECONDS
until ${BASH_CMD} -c "${CMD}" 2>&1 ; do
  echo ""
  if (( SECONDS - start_time >= TIMEOUT )); then
    echo "${msg} timed out! (output above)"
    exit 1
  fi
  echo "${msg} countdown: $((TIMEOUT - SECONDS + start_time)) "
  sleep 5
done
echo "${msg} done after $((SECONDS - start_time)) seconds."
```

You might use the function I gave like this:
```sh
# allow shorthand wait_for instead of full path
function wait_for() { ./script/dir/wait_for "$@"; }

wait_for "rook-ceph-mon-a-deploy pod to be ready" 60 "kubectl -n rook-ceph get pod | grep `rook-ceph-mon-a-deploy.*Ready'"
```

Or with a conditional like below, but I don't think you'll need this form:
```sh
./script/dir/wait_for "OSD provisioning to be complete" 900 "[[ \$(kubectl get -n rook-ceph pods 2>&1 | grep -c 'rook-ceph-osd-prepare.*Completed') -eq $num_osd_nodes ]]"
```

Reply:
Changed grep `rook-ceph-mon-a-debug` directly

Reply:
> A reusable function for waiting for a deployment's pod to be running is probably a good idea. I see this overall structure 3 times in this PR.
> 
> I use this script for my development environment tooling.
> 
> `wait_for`:
> 
> ```shell
> #!/usr/bin/env bash
> set -eEuo pipefail
> 
> REASON="$1"  # reason
> TIMEOUT="$2"  # in seconds
> CMD="$3"
> 
> BASH_CMD="${BASH_CMD:=bash}" # might be 'bash -x', for example
> 
> msg="Waiting ${TIMEOUT} seconds for ${REASON} ..."
> echo "${msg} countdown: ${TIMEOUT}"
> start_time=$SECONDS
> until ${BASH_CMD} -c "${CMD}" 2>&1 ; do
>   echo ""
>   if (( SECONDS - start_time >= TIMEOUT )); then
>     echo "${msg} timed out! (output above)"
>     exit 1
>   fi
>   echo "${msg} countdown: $((TIMEOUT - SECONDS + start_time)) "
>   sleep 5
> done
> echo "${msg} done after $((SECONDS - start_time)) seconds."
> ```
> 
> You might use the function I gave like this:
> 
> ```shell
> # allow shorthand wait_for instead of full path
> function wait_for() { ./script/dir/wait_for "$@"; }
> 
> wait_for "rook-ceph-mon-a-deploy pod to be ready" 60 \
>   "kubectl -n rook-ceph get pod | grep `rook-ceph-mon-a-deploy.*Ready'"
> ```
> 
> Or with a conditional like below, but I don't think you'll need this form:
> 
> ```shell
> ./script/dir/wait_for "OSD provisioning to be complete" 900 \
>   "[[ \$(kubectl get -n rook-ceph pods 2>&1 | grep -c 'rook-ceph-osd-prepare.*Completed') -eq $num_osd_nodes ]]"
> ```

There is only one-liner 
`   timeout 10 bash -c 'until [ "$(kubectl get pod $debug_osd -n test-cluster -o jsonpath='{.status.phase}'| grep -c "Running")" -eq 1 ]; do echo "waiting for pod rook-ceph-osd-0-debug to get created " && sleep 1; done'
`

The Above script gives a lot of information I would propose to add it in the rook CI as it gives us a lot of information

For a CI like Krew which is simpler and easy to go, I don't think we need such information for debugging.

Anyway, If you say I would be happy to add in Krew CI too. @BlaineEXE @subhamkrai thoughts?

Reply:
That's a pretty complicated one-liner. IMO, a reusable function is still good. When reading the code, you can see the intent without getting distracted by the boilerplate, and there will be less chance for errors in that boilerplate.

Reply:
> What's to stop this line from returning the old pod if it doesn't terminate immediately? Often, pods take a while to terminate. Instead, can we select just the debug pod? The name of the deployment is deterministic (`rook-ceph-mon-a-debug` IIRC) also, so we should be able to use that to our advantage.

Regarding this, it also makes me think that maybe we should wait for the old pod to be gone before creating the debug deployment. While a pod is terminating, we have to assume the OSD is still running, and if a user quickly starts into the debug pod and starts performing operations on an OSD while the OSD process is running, it could corrupt the disk. @travisn thoughts?

Reply:
I agree, we should first wait for older pod to remove before waiting for new one to start as pod takes some time to terminating and running again.